pkg/controller/common/readiness_checks.go

```diff
@@ -52,3 +53,10 @@ func IsDeploymentReady(deployment *v12.Deployment) (bool, error) {
 
 	return deployment.Status.ReadyReplicas == deployment.Status.Replicas, nil
 }
+
+func IsServiceReady(service *v1core.Service) bool {
+	if service == nil {
+		return false
+	}
+	return len(service.Status.LoadBalancer.Ingress) > 0
```

Comment:
Referring to this check in another comment, Any ideas on a better indicator of service readiness? (If we decide to keep the service readiness check?)

Reply:
Going to answer my own question: It's actually a pretty bad indicator and we probably don't need to use it at all.

Reply:
I don't know if service readiness is really a concern. Usually you create them and they become available immediately. If a correctly configured service fails then this might indicate problems with the internal DNS. And in that case you have bigger problems than the operator not able to communicate with Grafana. So i'd say a service readiness check is not required.